### PR TITLE
Raise minimum android sdk version

### PR DESCRIPTION
**Pull request by hawkrives**
- Update gradle
- Raise `compileSdkVersion` and `targetSdkVersion`
- Gradle complained about ndk splits, so I removed it
- Force all dependencies to use specified versions
- Use `implementation` instead of `compile`
- Gradle complained about spaces in the `rootProject.name`
- I think that's all

Does it build?

### CODE REVIEW

**Comment by rye** [comment]
I get a bunch of these
```console
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation'.
It will be removed at the end of 2018
WARNING: The specified Android SDK Build Tools version (23.0.1) is ignored, as it is below the minimum supported version (27.0.1) for Android Gradle Plugin 3.1.0-alpha09.
Android SDK Build Tools 27.0.1 will be used.
To suppress this warning, remove "buildToolsVersion '23.0.1'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
```
messages for the various dependencies when they configure.

Also, I can't install over the existing version (on my phone).  Local builds should always have a higher version number than the alphas, which should have higher numbers than the betas, which should have higher numbers than production?

**Comment by rye** [comment]
>Also, I can't install over the existing version. Local builds should always have a higher version number than the alphas, which should have higher numbers than the betas, which should have higher numbers than production?

This leads me to wonder if the real reason our builds are getting eaten (#2216) is perhaps that Play Store is trying to push a new version out but it fails, which would explain why it takes a while for the new release to come in.  Not sure.

**Comment by hawkrives** [comment]
@rye About installing – we always have increasing version numbers on the Play store, though, because we use the Circle build number. Locally, your build uses 1 as the build number, which is why it's not higher…

**Comment by hawkrives** [comment]
Following https://github.com/react-navigation/react-navigation/issues/1976, adding `android.enableAapt2=false` to gradle.properties fixes the production build by using the old asset processor.

It won't work forever, but it'll do for a while?

**Comment by hawkrives** [comment]
yay, android is fixed.

**Comment by rye** [comment]
I think there's probably a better way to generate our app version code to prevent the downgrading error message I get when I install on my phone, but I think this is good as long as it builds.  We could try a nightly?  Then a beta tomorrow, see if it eats it and whatnot.

**Comment by hawkrives** [comment]
Sure. I'll let you merge at your leisure. I'm going to go make a PR to disable Intel builds.

File: android/settings.gradle [hunk 1]
```diff
@@ -1,4 +1,4 @@
-rootProject.name = 'All About Olaf'
+rootProject.name = 'AllAboutOlaf'
```
Comment (rye) [inline android/settings.gradle:1]: Does this change our project on Play Store or anything human-readable by end users?  Or is this just internal to the build and APK itself?

File: android/app/build.gradle [hunk 2]
```diff
@@ -97,18 +97,14 @@ def enableSeparateBuildPerCPUArchitecture = true
 def enableProguardInReleaseBuilds = false
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "26.0.1"
+    compileSdkVersion 26
```
Comment (rye) [inline android/app/build.gradle:100]: Not sure why not 27?  It's out, and we have it in our CI build image.

File: android/settings.gradle (hunk 1 above)
Comment (hawkrives) [reply android/settings.gradle:1]: AFAIK, it's internal to the build? Gradle yelled at me that spaces would no longer be supported in Gradle 5, so I can't imagine that it'd be user-facing.

File: android/app/build.gradle (hunk 2 above)
Comment (hawkrives) [reply android/app/build.gradle:100]: Ah, because this is what Android Studio fixed it to for me.

File: android/settings.gradle (hunk 1 above)
Comment (rye) [reply android/settings.gradle:1]: Ah, okay.

File: android/app/build.gradle (hunk 2 above)
Comment (rye) [reply android/app/build.gradle:100]: Alright.  Well, we can ship like this for now.

**Review by rye — APPROVED** [review]
Very F-it- :shipit: at the moment, but really we can't tell if anything will go wrong without trying this out.

File: android/app/build.gradle (hunk 2 above)
Comment (hawkrives) [reply android/app/build.gradle:100]: I'm about to push and update to 27

Comment (hawkrives) [reply android/app/build.gradle:100]: Nevermind, not going to 27 tonight:

> GC overhead limit exceeded

**Pull request merged**